torch/fx/graph.py

```diff
@@ -487,6 +487,30 @@ def type_repr(o : Any):
                     type_repr(sub_type)
             return typename
 
+        # Set to keep track of all processed nodes for the purpose of use
+        # tracking and deleting values.
+        seen_nodes : Set[Node] = set()
+
+        def delete_unused_values(node : Node):
```

Comment:
The cheaper way to do this is to find the first use of the node in the reverse node list. This prevents having to do the subset checks which is u^2 where u is the node with the maximum number of uses.